src/Components/WebAssembly/DevServer/src/Server/Startup.cs

```diff
@@ -29,27 +30,28 @@ public static void Configure(IApplicationBuilder app, IConfiguration configurati
 
         app.UseWebAssemblyDebugging();
 
-        bool applyCopHeaders = configuration.GetValue<bool>("ApplyCopHeaders");
+        var webHostEnvironment = app.ApplicationServices.GetRequiredService<IWebHostEnvironment>();
+        var applyCopHeaders = configuration.GetValue<bool>("ApplyCopHeaders");
 
-        if (applyCopHeaders)
+        app.Use(async (ctx, next) =>
         {
-            app.Use(async (ctx, next) =>
+            if (ctx.Request.Path.StartsWithSegments("/_framework/blazor.boot.json"))
             {
-                if (ctx.Request.Path.StartsWithSegments("/_framework") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.server.js") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.web.js"))
+                ctx.Response.Headers.Append("Blazor-Environment", webHostEnvironment.EnvironmentName);
+            }
+            else if (applyCopHeaders && ctx.Request.Path.StartsWithSegments("/_framework") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.server.js") && !ctx.Request.Path.StartsWithSegments("/_framework/blazor.web.js"))
+            {
+                var fileExtension = Path.GetExtension(ctx.Request.Path);
+                if (string.Equals(fileExtension, ".js", StringComparison.Ordinal))
```

Comment:
Why didn't an analyzer catch this previously? Also, it's more consistent to ignore case.

```suggestion
                if (string.Equals(fileExtension, ".js", StringComparison.OrdinalIgnoreCase))
```

I can't think of a reason a request would come in with an uppercase `.JS` extension, but file extensions generally are case insensitive, and the dev server would still serve the correct file. `StartsWithSegments` also defaults to `OrdinalIgnoreCase`.

Reply:
There was a diagnostic for this, but the severity was "info" which is probably why it was missed initially.